### PR TITLE
Remove Express Server in favor of "nuxt generate"

### PR DESCRIPTION
**Pull request by smyth64**
Remove Express and the schulcloud-client submodule because we don't need it right now.

If it's needed again, the following has to be done:

```
git submodule add https://github.com/schul-cloud/schulcloud-client
```

add the file /server/index.js
```
const express = require("express");
const consola = require("consola");
const path = require("path");
const handlebars = require("handlebars");
const layouts = require("handlebars-layouts");
const handlebarsWax = require("handlebars-wax");
const { Nuxt, Builder } = require("nuxt");
const app = express();
const themeName = process.env.SC_THEME || "default";

const host = process.env.HOST || "localhost";
const port = process.env.PORT || 3000;

app.use(
	"/theme",
	express.static(path.join(__dirname, "../themes/" + themeName + "/static"))
);

/*
 * LECACY ROUTING
 */

// Set this path to the legacy schulcloud-client
const legacyClientRoot = path.join(__dirname, "../../schulcloud-client");

const handlebarsHelper = require(path.join(
	legacyClientRoot,
	"./helpers/handlebars"
));
const wax = handlebarsWax(handlebars)
	.partials(path.join(legacyClientRoot, "./views/**/*.{hbs,js}"))
	.helpers(layouts)
	.helpers(handlebarsHelper.helpers);

wax.partials(
	path.join(legacyClientRoot, `./theme/${themeName}/views/**/*.{hbs,js}`)
);

const viewDirs = [path.join(legacyClientRoot, "./views")];
viewDirs.unshift(path.join(legacyClientRoot, `./theme/${themeName}/views/`));

app.set("views", viewDirs);
app.engine("hbs", wax.engine);
app.set("view engine", "hbs");
app.use(express.static(path.join(legacyClientRoot, "./build/" + themeName)));
app.set("port", port);

// Import and Set Nuxt.js options
let config = require("./../../nuxt.config.js");
config.dev = !(process.env.NODE_ENV === "production");

// The legacy routings go here
setLegacyControllers([
	"about",
	"community",
	"help",
	"helpdesk",
	{
		route: "impressum",
		controller: "imprint",
	},
	"partner",
	"team",
]);

function setLegacyControllers(names) {
	for (const name of names) {
		if (typeof name === "object") {
			app.use(
				`/${name.route}/`,
				require(path.join(legacyClientRoot, `./controllers/${name.controller}`))
			);
		} else {
			app.use(
				`/${name}/`,
				require(path.join(legacyClientRoot, `./controllers/${name}`))
			);
		}
	}
}

async function start() {
	// Init Nuxt.js
	const nuxt = new Nuxt(config);

	// Build only in dev mode
	if (config.dev) {
		const builder = new Builder(nuxt);
		await builder.build();
	}

	// Give nuxt middleware to express
	app.use(nuxt.render);

	// Listen the server
	app.listen(port, host);
	consola.ready({
		message: `Server listening on http://${host}:${port}`,
		badge: true,
	});
}
start();
```

In the package.json, replace this line 
```
"dev:nuxt": "cross-env NODE_ENV=development nuxt"
```
with this one
```
"dev:nuxt": "cross-env NODE_ENV=development nodemon src/server/index.js --watch server"
```

### CODE REVIEW

**Comment by smyth64** [comment]
static files get now served correctly:
usage:
```
SC_THEME=n21 yarn build:nuxt
```

File: src/themes/default/config.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const images = "/theme/images/";
+const images = "/images/";
```
Comment (adrianjost) [inline src/themes/default/config.js]: It's the same base path for all themes and images, so you don't need this variable and include it directly in the path

File: src/themes/n21/config.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const images = "/theme/images/";
+const images = "/images/";
```
Comment (adrianjost) [inline src/themes/n21/config.js]: Same as mentioned above

File: package.json [hunk 3]
```diff
@@ -5,11 +5,11 @@
 	"author": "HPI Schul-Cloud",
 	"private": true,
 	"scripts": {
-		"dev:nuxt": "cross-env NODE_ENV=development nodemon src/server/index.js --watch server",
+		"dev:nuxt": "cross-env NODE_ENV=development nuxt",
 		"dev:docs": "vuepress dev docs",
 		"dev:storybook": "start-storybook -c stories/.storybook -p 6006",
 		"dev": "run-p dev:*",
-		"build:nuxt": "cross-env NODE_ENV=production nuxt generate",
+		"build:nuxt": "cross-env NODE_ENV=production nuxt generate && cross-env sh ./copy_static.sh",
```
Comment (adrianjost) [inline package.json]: I will test if this is working on windows as well

Comment (adrianjost) [reply package.json]: Works perfectly 👍

Comment (smyth64) [reply package.json]: image

**Pull request merged**